app/controllers/architectures_controller.rb

```diff
@@ -1,11 +1,28 @@
 class ArchitecturesController < ApplicationController
+  include ApplicationHelper
   include Foreman::Controller::AutoCompleteSearch
   include Foreman::Controller::Parameters::Architecture
 
   before_action :find_resource, :only => [:edit, :update, :destroy]
 
   def index
     @architectures = resource_base_search_and_page.includes(:operatingsystems)
+    respond_to do |format|
+      format.html do
+        render "index"
+      end
+      format.json do
+        render json: {
+          rows: rows(@architectures),
+          row_actions: row_actions,
+          columns: columns,
+          item_count: @architectures.count,
```

Comment:
How does this work with pagination? Does it not paginate at all or is this count just the paginated number?

Reply:
This should probably use `total_entries`.

Reply:
exactly, pagination will work only if the sum of `pages` & `per_page` items are lower than the `item_count`
also, if item_count is 0 we should display the empty state component

Reply:
Note, we have the `react_pagination_props` helper method. I'm wondering if we actually use this method anywhere.

Reply:
I also don't see it is used, it was part of the Audits page implementation, which seem to fetch its data from the Audits controller now in https://github.com/theforeman/foreman/blob/develop/app/controllers/audits_controller.rb#L10

Reply:
> exactly, pagination will work only if the sum of `pages` & `per_page` items are lower than the `item_count`

I think you may not have understood what I said. I think that `@architectures` is already paginated so the count is the paginated count. So this is probably correct (if I read @timogoebel's comment correctly):
```suggestion
          item_count: @architectures.total_entries,
```

Reply:
Alright, thanks for the clarification.. I saw that count is being used all over.. and indeed it returns the total amount of architectures, not just the paginated.